tests/phpunit/integration/Modules/Tag_ManagerTest.php

```diff
@@ -11,29 +11,89 @@
 namespace Google\Site_Kit\Tests\Modules;
 
 use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
 use Google\Site_Kit\Core\Modules\Module;
 use Google\Site_Kit\Core\Modules\Module_With_Service_Entity;
 use Google\Site_Kit\Core\Modules\Module_With_Owner;
 use Google\Site_Kit\Core\Modules\Module_With_Scopes;
 use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
 use Google\Site_Kit\Modules\Tag_Manager;
 use Google\Site_Kit\Modules\Tag_Manager\Settings;
 use Google\Site_Kit\Tests\Core\Modules\Module_With_Owner_ContractTests;
 use Google\Site_Kit\Tests\Core\Modules\Module_With_Scopes_ContractTests;
 use Google\Site_Kit\Tests\Core\Modules\Module_With_Service_Entity_ContractTests;
 use Google\Site_Kit\Tests\FakeHttp;
 use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\UserAuthenticationTrait;
+use Google\Site_Kit_Dependencies\Google\Service\TagManager\ContainerVersion;
+use Google\Site_Kit_Dependencies\Google\Service\TagManager\Tag;
 use Google\Site_Kit_Dependencies\GuzzleHttp\Promise\FulfilledPromise;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Request;
 use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Response;
 
 /**
  * @group Modules
  */
 class Tag_ManagerTest extends TestCase {
+	use UserAuthenticationTrait;
 	use Module_With_Scopes_ContractTests;
 	use Module_With_Owner_ContractTests;
 	use Module_With_Service_Entity_ContractTests;
 
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Options object.
+	 *
+	 * @var Options
+	 */
+	private $options;
+
+	/**
+	 * User object.
+	 *
+	 * @var WP_User
+	 */
+	private $user;
+
+	/**
+	 * User Options object.
+	 *
+	 * @var User_Options
+	 */
+	private $user_options;
+
+	/**
+	 * Authentication object.
+	 *
+	 * @var Authentication
+	 */
+	private $authentication;
+
+	/**
+	 * Tag Manager object.
+	 *
+	 * @var Tag_Manager
+	 */
+	private $tagmanager;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->context        = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options        = new Options( $this->context );
+		$this->user           = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		$this->user_options   = new User_Options( $this->context, $this->user->ID );
+		$this->authentication = new Authentication( $this->context, $this->options, $this->user_options );
+		$this->tagmanager     = new Tag_Manager( $this->context, $this->options, $this->user_options, $this->authentication );
+	}
```

Comment:
This is only really used for the one set of tests but it will run before all. Better to follow the current convention, otherwise we should update the rest to use these.